diff -up hdparm-9.16/Makefile.strictalias hdparm-9.16/Makefile
--- hdparm-9.16/Makefile.strictalias	2010-06-23 10:14:45.000000000 +0200
+++ hdparm-9.16/Makefile	2010-06-23 10:15:16.000000000 +0200
@@ -13,7 +13,7 @@ oldmandir = $(manprefix)/man
 CC ?= gcc
 STRIP ?= strip
 
-CFLAGS := -O2 -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs $(CFLAGS)
+CFLAGS := -O2 -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs -fno-strict-aliasing $(CFLAGS)
 
 LDFLAGS = -s
 #LDFLAGS = -s -static
